aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -135,15 +136,32 @@ Tensor angle(const Tensor& self) {
 }
 
 Tensor real(const Tensor& self) {
-  TORCH_CHECK(!self.is_complex(), "real is not yet implemented for complex tensors.");
-  return self;
+  if (self.is_complex()) {
+    auto size = self.sizes().vec();
+    if (self.numel() == 0) {
+      return at::empty({0}, self.options().dtype(c10::toValueType(self.scalar_type())));
+    } else {
+      auto float_tensor = at::native::view_complex_as_float(self);
```

Comment:
Looking at the implementation of `view_complex_as_float`, I don't think you want to use it here.
The returned Tensor does not own its memory. So as soon as the complex Tensor goes out of scope, the view will become invalid.
Since you return this view to the user, it should be owning it's memory or have a very very big warning in the doc:
```
def eval_my_fn(inp):
    complex_out = my_fn(inp)
    return complex_out.real #Only return the real part

out = eval_my_fn(inp)
print(out) # This will segfault.

Reply:
I think it's a good point and I will update the code to make the tensor from the same storage as self, but the case you mentioned above doesn't segfault with this code:
```
>>> def eval_my_fn(inp):
...     complex_out = inp.conj()
...     return complex_out.real
...
>>> out = eval_my_fn(torch.randn(2,2, dtype=torch.cfloat))
>>> print(out)
tensor([[-9.1000e-17,  5.6052e-45],
        [ 1.4013e-45,  0.0000e+00]])
```

Reply:
Most likely because you're lucky and the memory is not returned to the OS yet. But I'm sure that if you play around with sizes or reallocate a new Tensor with the same size you can get very bad stuff to happen.